docs/topics/plugins/versioning.md

```diff
@@ -0,0 +1,315 @@
+      "renderVersionsNavigationOnAllPages": true
+    }
+    """
+    pluginsMapConfiguration.set(
+            ["org.jetbrains.dokka.versioning.VersioningPlugin": versioningConfiguration]
+    )
+}
+```
+
+</tab>
+<tab title="Maven" group-key="mvn">
+
+```xml
+<plugin>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-maven-plugin</artifactId>
+    ...
+    <configuration>
+        <pluginsConfiguration>
+            <org.jetbrains.dokka.versioning.VersioningPlugin>
+                <version>1.5</version>
+                <versionsOrdering>
+                    <version>1.5</version>
+                    <version>1.4</version>
+                    <version>1.3</version>
+                    <version>1.2</version>
+                    <version>1.1</version>
+                    <version>alpha-2</version>
+                    <version>alpha-1</version>
+                </versionsOrdering>
+                <olderVersionsDir>${project.basedir}/documentation/version</olderVersionsDir>
+                <olderVersions>
+                    <version>${project.basedir}/documentation/alpha/alpha-2</version>
+                    <version>${project.basedir}/documentation/alpha/alpha-1</version>
+                </olderVersions>
+                <renderVersionsNavigationOnAllPages>true</renderVersionsNavigationOnAllPages>
+            </org.jetbrains.dokka.versioning.VersioningPlugin>
+        </pluginsConfiguration>
+    </configuration>
+</plugin>
+```
+
+</tab>
+<tab title="CLI" group-key="cli">
+
+```Bash
+java -jar dokka-cli-%dokkaVersion%.jar \
+     ...
+     -pluginsConfiguration "org.jetbrains.dokka.versioning.VersioningPlugin={\"version\": \"1.5\", \"versionsOrdering\": [\"1.5\", \"1.4\", \"1.3\", \"1.2\", \"1.1\", \"alpha-2\", \"alpha-1\"], \"olderVersionsDir\": \"documentation/version\", \"olderVersions\": [\"documentation/alpha/alpha-2\", \"documentation/alpha/alpha-1\"], \"renderVersionsNavigationOnAllPages\": true}"
+
+```
+
+Alternatively, via JSON configuration:
+```json
+{
+  "moduleName": "Dokka Example",
+  ...
+  "pluginsConfiguration": [
+    {
+      "fqPluginName": "org.jetbrains.dokka.versioning.VersioningPlugin",
+      "serializationFormat": "JSON",
+      "values": "{\"version\": \"1.5\", \"versionsOrdering\": [\"1.5\", \"1.4\", \"1.3\", \"1.2\", \"1.1\", \"alpha-2\", \"alpha-1\"], \"olderVersionsDir\": \"documentation/version\", \"olderVersions\": [\"documentation/alpha/alpha-2\", \"documentation/alpha/alpha-1\"], \"renderVersionsNavigationOnAllPages\": true}"
+    }
+  ]
+}
+```
+
+</tab>
+</tabs>
+
+## Generating versioned documentation
+
+With versioning plugin applied and configured, no other steps are needed: documentation can be built the usual way.
+
+Versioning plugin will add a `version.json` file to the output folder. This file will be used by the plugin to match
+versions and generate version navigation. If your previously generated documentation does not have that file, you
```

Comment:
Versioning plugin adds a template of the version navigation into HTML as well.

Reply:
Yeah, the idea was to use `version.json` as an indicator of whether documentation is versioned - not to say "add this file if it's not present"

Rephrased it a little, hopefully it's better now